tests/ui/PaginationTest.tsx

```diff
@@ -0,0 +1,348 @@
+        TestHelper.expectAPICommandToHaveBeenCalled('GetNewerActions', 0);
+    });
+
+    it('opens a chat and load older messages', async () => {
+        mockOpenReport(5, '8');
+        mockGetOlderActions(5);
+
+        await signInAndGetApp();
+        await navigateToSidebarOption(REPORT_ID);
+
+        expect(getReportActions()).toHaveLength(5);
+        TestHelper.expectAPICommandToHaveBeenCalled('OpenReport', 1);
+        TestHelper.expectAPICommandToHaveBeenCalledWith('OpenReport', 0, {reportID: REPORT_ID, reportActionID: ''});
+        TestHelper.expectAPICommandToHaveBeenCalled('GetOlderActions', 0);
+        TestHelper.expectAPICommandToHaveBeenCalled('GetNewerActions', 0);
+
+        // Scrolling here should trigger a new network request.
+        scrollToOffset(LIST_CONTENT_SIZE.height);
+        await waitForBatchedUpdatesWithAct();
+
+        TestHelper.expectAPICommandToHaveBeenCalled('OpenReport', 1);
+        TestHelper.expectAPICommandToHaveBeenCalled('GetOlderActions', 1);
+        TestHelper.expectAPICommandToHaveBeenCalledWith('GetOlderActions', 0, {reportID: REPORT_ID, reportActionID: '4'});
+        TestHelper.expectAPICommandToHaveBeenCalled('GetNewerActions', 0);
+
+        await waitForBatchedUpdatesWithAct();
+
+        // We now have 10 messages. 5 from the initial OpenReport and 3 from GetOlderActions.
+        // GetOlderActions only returns 3 actions since it reaches id '1', which is the created action.
+        expect(getReportActions()).toHaveLength(8);
+    });
+
+    it('opens a chat and load newer messages', async () => {
+        mockOpenReport(5, '5');
+        mockGetNewerActions(5);
+
+        await signInAndGetApp();
+        await navigateToSidebarOption(COMMENT_LINKING_REPORT_ID);
+
+        const link = screen.getByText('Link 1');
+        fireEvent(link, 'press');
+        await act(() => {
+            (NativeNavigation as NativeNavigationMock).triggerTransitionEnd();
+        });
+        // ReportScreen relies on the onLayout event to receive updates from onyx.
+        triggerListLayout();
+
+        expect(getReportActions()).toHaveLength(5);
+
+        // There is 1 extra call here because of the comment linking report.
+        TestHelper.expectAPICommandToHaveBeenCalled('OpenReport', 2);
+        TestHelper.expectAPICommandToHaveBeenCalledWith('OpenReport', 1, {reportID: REPORT_ID, reportActionID: '5'});
+        TestHelper.expectAPICommandToHaveBeenCalled('GetOlderActions', 0);
+        TestHelper.expectAPICommandToHaveBeenCalled('GetNewerActions', 0);
+
+        scrollToOffset(0);
+        await waitForBatchedUpdatesWithAct();
+
+        TestHelper.expectAPICommandToHaveBeenCalled('OpenReport', 2);
+        TestHelper.expectAPICommandToHaveBeenCalled('GetOlderActions', 0);
+        TestHelper.expectAPICommandToHaveBeenCalled('GetNewerActions', 1);
+        TestHelper.expectAPICommandToHaveBeenCalledWith('GetNewerActions', 0, {reportID: REPORT_ID, reportActionID: '5'});
```

Comment:
I want to make sure I'm understanding this. So we are opening a report by linking to an action within it. That calls OpenReport once for reportActionID: '5' and the mock returns 5 actions. Then we scroll to the bottom to load 5 new actions. Why is the same ID passed? Wouldn't the ID be 10?

Reply:
This one is the forward pagination case so it is a little different.

- OpenReport call with id '5' will load ids ['5', '4', '3', '2', '1']
- Scroll to start
- GetNewerActions is called with id '5' and will load ['5', '6', '7', '8', '9', '10']

The ids are generated chronologically so newer messages always have bigger id.

When passing a reportActionID to OpenReport and GetOlderActions it returns that report and older reports (meaning ids will be lowering) and GetNewerActions it returns that report and newer report (meaning ids will be growing).

Reply:
Ah ok, now it makes sense. I had assumed opening a report at a given ID would return IDs after the given one. Thank you for explaining.